src/init/features/dataconnect/index.ts

```diff
@@ -18,20 +20,159 @@ const SCHEMA_TEMPLATE = readFileSync(join(TEMPLATE_ROOT, "schema.gql"), "utf8");
 const QUERIES_TEMPLATE = readFileSync(join(TEMPLATE_ROOT, "queries.gql"), "utf8");
 const MUTATIONS_TEMPLATE = readFileSync(join(TEMPLATE_ROOT, "mutations.gql"), "utf8");
 
+interface RequiredInfo {
+  serviceId: string;
+  locationId: string;
+  cloudSqlInstanceId: string;
+  cloudSqlDatabase: string;
+  connectorId: string;
+  isNewInstance: boolean;
+  isNewDatabase: boolean;
+}
 export async function doSetup(setup: Setup, config: Config): Promise<void> {
-  if (setup.projectId) {
-    await ensureApis(setup.projectId);
+  let info: RequiredInfo = {
+    serviceId: "",
+    locationId: "",
+    cloudSqlInstanceId: "",
+    isNewInstance: false,
+    cloudSqlDatabase: "",
+    isNewDatabase: false,
+    connectorId: "default-connector",
+  };
+  info = await promptForService(setup, info);
+
+  if (info.cloudSqlInstanceId === "") {
+    info = await promptForCloudSQLInstance(setup, info);
   }
-  const serviceId = await promptOnce({
-    message: "What ID would you like to use for this service?",
+
+  if (info.cloudSqlDatabase === "") {
+    info = await promptForDatabase(setup, config, info);
+  }
+
+  const defaultConnectionString =
```

Comment:
Okay, to be a TODO, but eventually we will remove this to a config or prompt when emulator starts.